types/service/database_service_configuration.go

```diff
@@ -233,27 +375,57 @@ type AwsRdsIamAuthConfiguration struct {
 	CaCertificate  string                `json:"ca_certificate,omitempty"`
 }
 
-// GoogleCloudSqlUsernameAndPasswordAuthConfiguration represents auth configuration for Google Cloud SQL databases that
+func (config AwsRdsIamAuthConfiguration) Validate() error {
+	if config.Username == "" {
+		return errors.New("username is required")
+	}
+	if err := config.AwsCredentials.Validate(); err != nil {
+		return fmt.Errorf("invalid AWS credentials: %w", err)
+	}
+	return nil
+}
+
+// GcpCloudSqlUsernameAndPasswordAuthConfiguration represents auth configuration for Google Cloud SQL databases that
 // use username and password for authentication, and are connected to the upstream using the Cloud SQL Connector.
 // You must provide a username, a password, an Cloud SQL instance ID and Google credentials that are copied from the JSON
 // credentials file.
-type GoogleCloudSqlUsernameAndPasswordAuthConfiguration struct {
+type GcpCloudSqlUsernameAndPasswordAuthConfiguration struct {
 	UsernameAndPassword
 	InstanceId         string `json:"instance_id"`
 	GcpCredentialsJson string `json:"gcp_credentials_json"`
 }
 
-// GoogleCloudSqlIamAuthConfiguration represents auth configuration for Google Cloud SQL databases that use IAM authentication,
+func (config GcpCloudSqlUsernameAndPasswordAuthConfiguration) Validate() error {
```

Comment:
Godoc here and a couple new places as well.  So far the convention for these is just 

`Validate validates the GcpCloudSqlUsernameAndPasswordAuthConfiguration`